File: mobile/src/main/java/net/mm2d/dmsexplorer/viewmodel/helper/PipHelpers.java
/*
 * Copyright (c) 2017 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.dmsexplorer.viewmodel.helper;

import android.app.Activity;
import android.os.Build;

import androidx.annotation.NonNull;

/**
 * @author <a href="mailto:ryo@mm2d.net">大前良介 (OHMAE Ryosuke)</a>
 */
public class PipHelpers {
    private static final MovieActivityPipHelper MOVIE_MOCK = new MovieActivityPipHelperEmpty();

    public static boolean isSupported() {
        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.O;
    }

    @NonNull
    public static MovieActivityPipHelper getMovieHelper(@NonNull final Activity activity) {
        if (isSupported()) {
            return new MovieActivityPipHelperOreo(activity);
        }
        return MOVIE_MOCK;
    }
}

File: mobile/src/main/java/net/mm2d/dmsexplorer/viewmodel/helper/PipHelpers.kt
/*
 * Copyright (c) 2017 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.dmsexplorer.viewmodel.helper

import android.app.Activity
import android.content.Context
import android.content.pm.PackageManager
import android.os.Build

/**
 * @author [大前良介 (OHMAE Ryosuke)](mailto:ryo@mm2d.net)
 */
object PipHelpers {
    private val MOCK = MovieActivityPipHelperEmpty()

    fun isSupported(context: Context): Boolean = Build.VERSION.SDK_INT >= Build.VERSION_CODES.O &&
            context.packageManager.hasSystemFeature(PackageManager.FEATURE_PICTURE_IN_PICTURE)

    fun getMovieHelper(activity: Activity): MovieActivityPipHelper =
        if (isSupported(activity)) MovieActivityPipHelperOreo(activity) else MOCK
}
